scripts/test_matrix.py

```diff
@@ -0,0 +1,190 @@
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+import sys
+
+prov_all 	= {'ofi-rxm': 1, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 1, 'verbs': 1}
+prov_msg 	= {'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 1}
+prov_rdm 	= {'ofi-rxm': 1, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 1}
+prov_dgram 	= {'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 1, 'verbs': 0}
+
+av_test = "av_test -g GOOD_ADDR -n 1 -s SERVER_ADDR"
+
+def sh_ctx(ep=None, opt=None):
+	cmd = "shared_ctx"
+	if ep:
+		cmd += " -e " + ep
+	if opt:
+		if opt == "notx":
+			cmd += " --no-tx-shared-ctx"
+		elif opt == "norx":
+			cmd += " --no-rx-shared-ctx"
+		else:
+			print "Unknown shared_ctx test opt"
+			exit(1)
+	return  cmd
+
+unit = [
+	['getinfo_test',	prov_all],
+	# av_test asks for FI_SOCKADDR which is not supported by psm and psm2
+	[av_test,		{'ofi-rxm': 1, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 1, 'verbs': 1}],
+	['dom_test -n 2', 	prov_all],
+	['eq_test', 		prov_all],
+	['cq_test', 		prov_all],
+	['mr_test', 		prov_all],
+	['cntr_test', 		prov_all],
+]
+
+neg_unit = [
+	['dgram g00n13s', prov_dgram],
+	['rdm g00n13s' 	, prov_rdm],
+	['msg g00n13s'	, prov_msg],
+]
+
+simple = [
+	['cm_data', 			prov_msg],
+	# cq_data test supports only FI_EP_MSG
+	['cq_data', 			prov_msg],
+	['dgram', 			prov_dgram],
+	['dgram_waitset', 		prov_dgram],
+	['msg', 			prov_msg],
+	['msg_epoll', 			prov_msg],
+	['msg_sockets', 		prov_msg],
+	['poll -t queue', 		{'ofi-rxm': 1, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['poll -t counter',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm',				prov_rdm],
+	# rdm_rma_simple requires FI_RMA_EVENT
+	['rdm_rma_simple',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_rma_trigger',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg"),			{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg","notx"),		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg","norx"),		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm"),			{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm","notx"),		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm","norx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram"),		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram","notx"),	{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram","norx"),	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_tagged_peek',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	# psm2 supports scalable endpoints but the test doesn't support FI_CONTEXT
```

Comment:
This is already encoded in the output of `fi_info`. Instead of programming the assumptions into the script, maybe this script should map features to tests and build the matrix based on the output of `fi_info`.